tests/test_schema.py

```diff
@@ -406,6 +406,30 @@ def test_schema_annotation_inheritance(self):
         self.assertEqual(
             Object2.get_annotation(schema, 'test::inh'), 'inherit me')
 
+    def test_schema_annotation_inheritance_02(self):
```

Comment:
Is there a reason why we need both of the tests in this PR? They seem to be kinda doing the same thing: checking the schema for annotations on the derived type. What is the thing that one of them tests that the other doesn't?

Reply:
They are largely equivalent, the difference is in how we check the effects.